sycl/include/sycl/accessor.hpp

```diff
@@ -2052,17 +2052,20 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   bool empty() const noexcept { return size() == 0; }
 
   template <int Dims = Dimensions, typename = detail::enable_if_t<(Dims > 0)>>
-  range<Dimensions> get_range() const {
-    return detail::convertToArrayOfN<Dimensions, 1>(getAccessRange());
+  range<Dims> get_range() const {
+    if constexpr (Dimensions == 0)
```

Comment:
Should there also be "Dims" then? Though I don't know why we need it at all.

Reply:
The reason it's done so, because 0-dim range is not allowed to be constructed. If we do not have  that condition or use `Dimension` instead of `Dim` then we'll bump into a `static_assert` for construction 0-dim range.
If you look into the other changed places, I have passed `AdjustedDim` which is 1 for 0-dim accessors.

Reply:
Yeah, I mean `Dimensions => Dims` change as you did it here

Reply:
Or they are different?

Reply:
They are different when one would like to get range with different dim compared to class's dim, similar to current scenario. 
I was reluctant to use `AdjustedDim` for this function, mainly due to performance, as it would need to call function in `else` part, with the answer we already know.

Reply:
In what cases do we have uses of `get_range` with a different range? That isn't allowed by the spec, so there shouldn't be such cases. I suspect the reason why we have the above dependence to be able to not define it in the 0-dim case.

Another option is here is to change the requirements above to something like

```c++
template <int Dims = Dimensions, typename = std::enable_if_t<Dims == Dimensions && (Dims > 0)>>
```

Arguably the same should be applied to `get_offset` below.

Reply:
thanks @steffenlarsen.